### PR TITLE
Lockdown Microsoft.ML.Data Dataview folder.

### LINKED ISSUE #2606: Internalization of Microsoft.ML.Data Dataview folder
https://github.com/dotnet/machinelearning/blob/b881fdc2eab55e184eb0856d4c10c732a58e0887/src/Microsoft.ML.Data/DataView/AppendRowsDataView.cs#L30

https://github.com/dotnet/machinelearning/blob/b881fdc2eab55e184eb0856d4c10c732a58e0887/src/Microsoft.ML.Data/DataView/CompositeRowToRowMapper.cs#L14

https://github.com/dotnet/machinelearning/blob/b881fdc2eab55e184eb0856d4c10c732a58e0887/src/Microsoft.ML.Data/DataView/DataViewConstructionUtils.cs#L727

https://github.com/dotnet/machinelearning/blob/b881fdc2eab55e184eb0856d4c10c732a58e0887/src/Microsoft.ML.Data/DataView/DataViewConstructionUtils.cs#L755

https://github.com/dotnet/machinelearning/blob/b881fdc2eab55e184eb0856d4c10c732a58e0887/src/Microsoft.ML.Data/DataView/EmptyDataView.cs#L15

https://github.com/dotnet/machinelearning/blob/b881fdc2eab55e184eb0856d4c10c732a58e0887/src/Microsoft.ML.Data/DataView/OpaqueDataView.cs#L15

https://github.com/dotnet/machinelearning/blob/b881fdc2eab55e184eb0856d4c10c732a58e0887/src/Microsoft.ML.Data/DataView/ZipDataView.cs#L17

need to be moved to somewhere else (We probably already have schema utils):
https://github.com/dotnet/machinelearning/blob/b881fdc2eab55e184eb0856d4c10c732a58e0887/src/Microsoft.ML.Data/DataView/SimpleRow.cs#L73

I don't see any usages, need to check is it still relevant:
https://github.com/dotnet/machinelearning/blob/b881fdc2eab55e184eb0856d4c10c732a58e0887/src/Microsoft.ML.Data/DataView/LambdaFilter.cs#L19 

not sure about this:

https://github.com/dotnet/machinelearning/blob/b881fdc2eab55e184eb0856d4c10c732a58e0887/src/Microsoft.ML.Data/DataView/ArrayDataViewBuilder.cs#L17

https://github.com/dotnet/machinelearning/blob/b881fdc2eab55e184eb0856d4c10c732a58e0887/src/Microsoft.ML.Data/DataView/TypedCursor.cs#L35

https://github.com/dotnet/machinelearning/blob/b881fdc2eab55e184eb0856d4c10c732a58e0887/src/Microsoft.ML.Data/DataView/TypedCursor.cs#L45

### PR DESCRIPTION
**Pull request by codemzs**
fixes #2606

### CODE REVIEW

File: src/Microsoft.ML.Data/DataView/LambdaFilter.cs [hunk 1]
```diff
@@ -1,194 +0,0 @@
...
-using Microsoft.ML.Data.Conversion;
-using Microsoft.ML.Model;
-
-namespace Microsoft.ML.Data
-{
-    /// <summary>
-    /// This applies the user provided RefPredicate to a column and drops rows that map to false. It automatically
-    /// injects a standard conversion from the actual type of the source column to typeSrc (if needed).
-    /// </summary>
-    public static class LambdaFilter
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/DataView/LambdaFilter.cs]: >LambdaFilter [](start = 24, length = 12)

I don't think we should delete this. While I doubt we'll get around to this for v1, I was planning on using this to solve #1763.

Comment (codemzs) [reply src/Microsoft.ML.Data/DataView/LambdaFilter.cs]: Ok, thanks. I will revert it.

File: src/Microsoft.ML.Data/DataView/SimpleRow.cs [hunk 2]
```diff
@@ -70,7 +70,8 @@ public override bool IsColumnActive(int col)
         }
     }
 
-    public static class SimpleSchemaUtils
+    [BestFriend]
+    internal static class SimpleSchemaUtils
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/DataView/SimpleRow.cs]: >SimpleSchemaUtils [](start = 26, length = 17)

Now that creating a schema using the schema builder is much easier than it was back when this was much easier, what do we think about just deleting this outright and replacing the five usages?

File: src/Microsoft.ML.Data/DataView/LambdaFilter.cs (hunk 1 above)
Comment (TomFinley) [reply src/Microsoft.ML.Data/DataView/LambdaFilter.cs]: All right, can still be internal BTW till we work out its public surface, but I'd rather not delete.

File: src/Microsoft.ML.Data/DataView/SimpleRow.cs (hunk 2 above)
Comment (codemzs) [reply src/Microsoft.ML.Data/DataView/SimpleRow.cs]: Sounds lovely!

**Pull request merged**